tests/conversion/viewconversiondispatcher.js

```diff
@@ -207,91 +195,169 @@ describe( 'ViewConversionDispatcher', () => {
-				data.output = conversionApi.convertChildren( data.input, consumable );
+		describe( 'convertChildren', () => {
+			it( 'should fire conversion for all children of passed element and return conversion results wrapped in document fragment', () => {
+				silenceWarnings();
+
+				dispatcher.on( 'documentFragment', ( evt, data, consumable, conversionApi ) => {
+					spy();
+
+					const result = conversionApi.convertChildren( data.input, consumableMock, data );
+
+					expect( result ).to.be.instanceof( ModelDocumentFragment );
+					expect( result.childCount ).to.equal( 2 );
+					expect( result.getChild( 0 ) ).to.equal( modelP );
+					expect( result.getChild( 1 ) ).to.equal( modelText );
+				} );
+
+				dispatcher.convert( new ViewDocumentFragment( [ viewP, viewText ] ), { foo: 'bar' } );
+
+				expect( spy.calledOnce ).to.be.true;
+				expect( spyP.calledOnce ).to.be.true;
+				expect( spyText.calledOnce ).to.be.true;
 			} );
 
-			const viewStructure = new ViewContainerElement( 'div', null, [
-				new ViewContainerElement( 'p', null, [
-					new ViewContainerElement( 'span', { class: 'nice' }, [
-						new ViewAttributeElement( 'a', { href: 'foo.html' }, new ViewText( 'foo' ) ),
-						new ViewText( ' bar ' ),
-						new ViewAttributeElement( 'i', null, new ViewText( 'xyz' ) )
-					] )
-				] ),
-				new ViewContainerElement( 'p', null, [
-					new ViewAttributeElement( 'strong', null, [
-						new ViewText( 'aaa ' ),
-						new ViewAttributeElement( 'span', null, new ViewText( 'bbb' ) ),
-						new ViewText( ' ' ),
-						new ViewAttributeElement( 'a', { href: 'bar.html' }, new ViewText( 'ccc' ) )
-					] )
-				] )
-			] );
-
-			expect( dispatcher.convert( viewStructure ) ).to.deep.equal( [ 'foo', ' bar ', 'xyz', 'aaa ', 'bbb', ' ', 'ccc' ] );
+			it( 'should filter out incorrectly converted elements and log warnings', () => {
+				sinon.spy( log, 'warn' );
+
+				dispatcher.on( 'documentFragment', ( evt, data, consumable, conversionApi ) => {
+					spy();
+
+					const result = conversionApi.convertChildren( data.input, consumableMock, data );
+
+					expect( result ).to.be.instanceof( ModelDocumentFragment );
+					expect( result.childCount ).to.equal( 2 );
+					expect( result.getChild( 0 ) ).to.equal( modelP );
+					expect( result.getChild( 1 ) ).to.equal( modelText );
+				} );
+
+				dispatcher.convert( new ViewDocumentFragment( [ viewArray, viewP, viewDiv, viewText, viewNull ] ), { foo: 'bar' } );
+
+				expect( spy.calledOnce ).to.be.true;
+				expect( spyNull.calledOnce ).to.be.true;
+				expect( spyArray.calledOnce ).to.be.true;
+				expect( log.warn.calledOnce ).to.be.true;
+
+				log.warn.restore();
+			} );
 		} );
 	} );
+
+	// Silences warnings that pop up in tests. Use when the test checks a specific functionality and we are not interested in those logs.
+	// No need to restore `log.warn` - it is done in `afterEach()`.
+	function silenceWarnings() {
```

Comment:
I'm not 100% sure but I'm afraid that this may silence too many warnings as `log` may be defined just once at the beginning of this file. And if we ever switch to bundling all test files into one module (to improve Webpack's performance) this will silence all logs forever.